doc/guides/west/manifest.rst

```diff
@@ -775,63 +772,62 @@ hosted at ``https://git.example.com/mainline/manifest``.
      projects:
        - name: app
          url: https://git.example.com/mainline/app
-       - name: library
-         url: https://git.example.com/mainline/library
-         revision: refs/heads/only-in-mainline
-       - name: library2
-         url: https://git.example.com/mainline/library-2
-       - name: unnecessary-project
-         url: https://git.example.com/mainline/unnecessary-project
+       - name: lib
+         path: libraries/lib
+         url: https://git.example.com/mainline/lib
+       - name: lib2
+         path: libraries/lib2
+         url: https://git.example.com/mainline/lib2
 
    # downstream west.yml:
    manifest:
      projects:
        - name: mainline
          url: https://git.example.com/mainline/manifest
          import:
-           whitelist:
-             - library2
+           name-whitelist:
              - app
+             - lib2
            rename:
              app: mainline-app
-       - name: library2
-         path: mainline-lib2
        - name: app
          url: https://git.example.com/downstream/app
-       - name: library
-         url: https://git.example.com/downstream/library
+       - name: lib3
+         path: libraries/lib3
+         url: https://git.example.com/downstream/lib3
 
 An equivalent manifest in a single file would be:
 
 .. code-block:: yaml
 
    manifest:
      projects:
-       - name: library2
-         path: mainline-lib2
-         url: https://git.example.com/mainline/library-2
        - name: mainline-app
          url: https://git.example.com/mainline/app
+       - name: lib2
+         path: libraries/lib2
+         url: https://git.example.com/mainline/lib2
        - name: mainline
          url: https://git.example.com/mainline/manifest
        - name: app
          url: https://git.example.com/downstream/app
-       - name: library
-         url: https://git.example.com/downstream/library
+       - name: lib3
+         path: libraries/lib3
+         url: https://git.example.com/downstream/lib3
 
-If a whitelist had not been used:
+If a whitelist had not been used, the ``lib2`` project from the mainline
+manifest would have been imported.
 
-- The ``library`` project in the final manifest would have had its ``revision``
-  set to ``refs/heads/only-in-mainline``.
-- ``unnecessary-project`` would have been imported.
+If a rename had not been used, the downstream manifest's ``app`` project would
+have shadowed the mainline manifest's project by the same name.
```

Comment:
Is it clear that `rename` also changes the path of the imported project ?
And what would have happened in those cases ?:
```
# mainline west.yml:
manifest:
  projects:
    - name: app
      path: app-path
      url: https://git.example.com/mainline/app

# downstream west.yml:
manifest:
  projects:
    - name: mainline
      url: https://git.example.com/mainline/manifest
      import:
        name-whitelist:
          - app
          - lib2
        rename:
          app: mainline-app
    - name: app
      url: https://git.example.com/downstream/app
```
Will `mainline-app` be placed in path `app-path`, or in path `mainline-app`.

or this case, where both projects are defined with colliding paths.
```
# mainline west.yml:
manifest:
  projects:
    - name: app
      path: app-colliding-path
      url: https://git.example.com/mainline/app

# downstream west.yml:
manifest:
  projects:
    - name: mainline
      url: https://git.example.com/mainline/manifest
      import:
        name-whitelist:
          - app
          - lib2
        rename:
          app: mainline-app
    - name: app
      path: app-colliding-path
      url: https://git.example.com/downstream/app
```

Reply:
> Is it clear that rename also changes the path of the imported project ?

Looking at code, it seems path is not renamed, if set in mainline manifest, is that correct ?
And in that case, what happens if there is a path collision ?

Reply:
I think I am just going to remove support for `rename` for now.